pkg/bundle/controller.go

```diff
@@ -172,8 +186,8 @@ func AddBundleController(ctx context.Context, mgr manager.Manager, opts Options)
 
 		// Watch Secrets in trust Namespace. Only cache metadata.
 		// Reconcile Bundles who reference a modified source Secret.
-		Watches(source.NewKindWithCache(new(corev1.Secret), sourceCache), handler.EnqueueRequestsFromMapFunc(
-			func(obj client.Object) []reconcile.Request {
+		WatchesRawSource(&source.Informer{Informer: secretInformer}, handler.EnqueueRequestsFromMapFunc(
+			func(_ context.Context, obj client.Object) []reconcile.Request {
```

Comment:
question: here we ignore the context that's passed to this function in favour of the one that's in scope (see e.g. `bundleList := b.mustBundleList(ctx)`).

Why not use this context instead? It seems possible that the controller-runtime logic which calls this function could modify the context in some way.

```suggestion
			func(ctx context.Context, obj client.Object) []reconcile.Request {
```

Reply:
Thanks, didn't notice this one was actually using context!